### PR TITLE
[WIP] link: improve JAWS readability

### LINKED ISSUE #1476: link: improve JAWS readability
JAWS should read
- [x] `link` as "link"
- [x] `link_pseudo` as "button"
- [x] `link_disabled` as "link unavailable"
- [x] `content` field
- [x] `title` special field

Check and fix divergencies.

### PR DESCRIPTION
**Pull request by aristov**
closes #1476

### CODE REVIEW

File: common.blocks/link/link.bemhtml [hunk 1]
```diff
@@ -15,10 +15,14 @@ block('link')(
 
     attrs()(function() {
         var ctx = this.ctx,
-            attrs = {},
+            isDisabled = !!this.mods.disabled,
+            attrs = {
+                role : 'link',
```
Comment (aristov) [inline common.blocks/link/link.bemhtml]: TODO: avoid permanent `role="link"`

File: common.blocks/link/link.bh.js [hunk 2]
```diff
@@ -8,10 +8,14 @@ module.exports = function(bh) {
         var url = typeof json.url === 'object'? // url could contain bemjson
                 bh.apply(json.url) :
                 json.url,
-            attrs = {},
+            isDisabled = !!ctx.mod('disabled'),
+            attrs = {
+                role : 'link',
+                'aria-disabled' : isDisabled
```
Comment (aristov) [inline common.blocks/link/link.bh.js]: TODO: add `aria-disabled` attribute only if it is `true`

File: common.blocks/link/link.js [hunk 3]
```diff
@@ -26,12 +26,16 @@ provide(BEMDOM.decl({ block : this.name, baseBlock : Control }, /** @lends link.
...
+                    .removeAttr('href')
+                    .attr('aria-disabled', true);
             },
 
             '' : function() {
                 this.__base.apply(this, arguments);
-                this.domElem.attr('href', this._url);
+                this.domElem
+                    .attr('href', this._url)
+                    .attr('aria-disabled', false);
```
Comment (aristov) [inline common.blocks/link/link.js]: TODO: `.removeAttr(...`

File: common.blocks/link/link.bemhtml (hunk 1 above)
Comment (aristov) [inline common.blocks/link/link.bemhtml]: @veged
From [specification](http://www.w3.org/TR/wai-aria/roles):

> Roles are element types and authors MUST NOT change role values over time or with user actions.

Comment (veged) [inline common.blocks/link/link.bemhtml]: а повтори ещё раз, в каких случаях нам нужен `role="link"`?

Comment (aristov) [inline common.blocks/link/link.bemhtml]: Для `_disabed`. В устном обсуждении договорились, что добавление ссылке атрибута `role="link"` не является изменением роли, это не противоречит спецификации, поэтому TODO в силе.

**Comment by aristov** [comment]
need rebase

**Comment by sipayRT** [comment]
@aristov I propose to create a separate branch for a11y-PR's (for example, issues/1206v2 - main issue about a11y) and merge them(PR's) to it. And when all issues will be closed we will merge main branch to v2. What do you think about it?

**Comment by sipayRT** [comment]
@aristov ping

**Comment by sipayRT** [comment]
moved to https://github.com/bem/bem-components/pull/1555

**Comment by aristov** [comment]
> @aristov I propose to create a separate branch for a11y-PR's (for example, issues/1206v2 - main issue about a11y) and merge them(PR's) to it. And when all issues will be closed we will merge main branch to v2. What do you think about it?

I thought about the same idea, but couldn't see the reason why to do so. I think this is the complication out of the blue. Just more hell with branch rebasing and no profit.

**Comment by sipayRT** [comment]
@aristov at least, we can test all changes **before** merging to v2. Rebasing — it's no so difficult thing ;)

**Pull request closed without merging**